python/cudf/cudf/core/index.py

```diff
@@ -450,6 +451,15 @@ def __getitem__(self, index):
             return self.start + index * self.step
         return self._as_int_index()[index]
 
+    def _get_columns_by_label(self, labels) -> Index:
+        # used in .sort_values
+        if isinstance(labels, Hashable) and labels == self.name:
+            return self._as_int_index()
+        elif list(self.names) == list(labels):
```

Comment:
I think CI is failing on this example `df = cudf.DataFrame([1], columns=cudf.Index(np.array([1], dtype=np.int8))).sort_values(0)`. You get a TypeError from `list(labels)`.

Should this check if `labels` are in `self.names`?

Reply:
Thanks I was able to fix this.

The issue here is that `labels` can be scalar-like or list-like so the equality matching needs to fit both cases